aerogear/data-sync-server
# aerogear-8024: use utils from the core package

https://issues.jboss.org/browse/AEROGEAR-8024

Data sync server now accepts a `data-sync-client-info` header which is Base64 decoded and it prints whatever in that to audit logs.

This is to be done to have the client/app/device information in audit logs so that we can create aggregations/searches for the client info.

Work done:

* Adapt for the changes in https://github.com/aerogear/data-sync-gql-core/pull/14
* Return 400 if the client info is malformed.

How to verify:

1. Do the npm link stuff on core lib to have these changes in data-sync-server
2. Start data sync server
3. Open localhost:8000/graphql to access playground
4. Try querying (e.g. `{allMemes{id}}`) with these headers (can be added using the "HTTP HEADERS" button in the bottom):
4.1. No header --> no `clientInfo` field in audit logs
4.2. `{"data-sync-client-info":""}` --> no `audit.clientInfo` field in audit logs
4.3. `{"data-sync-client-info":"foo"}` --> should receive 400
4.4. `{"data-sync-client-info":"eyJhIjoxfQ=="}` --> `audit.clientInfo` field in audit logs with content `{"a":1}`
4.5. `{"data-sync-client-info":"eyJhIjoxfQfoo=="}` --> should receive 400
5. run `npm test`

Also note that the actual log is now wrapped in an `audit` field:
```
    tag: "AUDIT"
    audit: {
      "msg": "",
      "requestId": 34,
      "operationType": "query",
      "parentTypeName": "Query",
      "path": "allMemes",
      "success": true,
      "arguments": {},
      "dataSourceType": "InMemory",
      "clientInfo": {
        "a": 1
      }
    }
```
